### ISSUE TITLE
skipping: WithBacktrace { inner: Msg("unsupported storage type LongStorage") ... }

### ISSUE THREAD

**Issue by liebkne** [huggingface/candle#1151]
Hi, I get the below trace when I change the `mmap` loader to `pth` loader in the BERT example. 

The example is successfully built and the output seems the same as the origin example. Does the trace matter?

Thanks.

Trace:

```
skipping: WithBacktrace { inner: Msg("unsupported storage type LongStorage"), backtrace: Backtrace [{ fn: "std::backtrace_rs::backtrace::libunwind::trace", file: "/rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/../../backtrace/src/backtrace/libunwind.rs", line: 93 }, { fn: "std::backtrace_rs::backtrace::trace_unsynchronized", file: "/rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/../../backtrace/src/backtrace/mod.rs", line: 66 }, { fn: "std::backtrace::Backtrace::create", file: "/rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/backtrace.rs", line: 331 }, { fn: "std::backtrace::Backtrace::capture", file: "/rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/backtrace.rs", line: 297 }, { fn: "candle_core::error::Error::bt", file: "/Users/user/.cargo/git/checkouts/candle-0c2b4fa9e5801351/b43ab6c/candle-core/src/error.rs", line: 218 }, { fn: "candle_core::pickle::rebuild_args", file: "/Users/user/.cargo/git/checkouts/candle-0c2b4fa9e5801351/b43ab6c/candle-core/src/pickle.rs", line: 613 }, { fn: "candle_core::pickle::Object::into_tensor_info", file: "/Users/user/.cargo/git/checkouts/candle-0c2b4fa9e5801351/b43ab6c/candle-core/src/pickle.rs", line: 229 }, { fn: "candle_core::pickle::read_pth_tensor_info", file: "/Users/user/.cargo/git/checkouts/candle-0c2b4fa9e5801351/b43ab6c/candle-core/src/pickle.rs", line: 670 }, { fn: "candle_core::pickle::PthTensors::new", file: "/Users/user/.cargo/git/checkouts/candle-0c2b4fa9e5801351/b43ab6c/candle-core/src/pickle.rs", line: 691 }, { fn: "candle_nn::var_builder::VarBuilderArgs<alloc::boxed::Box<dyn candle_nn::var_builder::SimpleBackend>>::from_pth", file: "/Users/user/.cargo/git/checkouts/candle-0c2b4fa9e5801351/b43ab6c/candle-nn/src/var_builder.rs", line: 478 }, { fn: "chain_embedding_rs::Args::build_model_and_tokenizer", file: "./src/main.rs", line: 77 }, { fn: "chain_embedding_rs::main", file: "./src/main.rs", line: 99 }, { fn: "core::ops::function::FnOnce::call_once", file: "/rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/core/src/ops/function.rs", line: 250 }, { fn: "std::sys_common::backtrace::__rust_begin_short_backtrace", file: "/rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/sys_common/backtrace.rs", line: 154 }, { fn: "std::rt::lang_start::{{closure}}", file: "/rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/rt.rs", line: 166 }, { fn: "core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once", file: "/rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/core/src/ops/function.rs", line: 284 }, { fn: "std::panicking::try::do_call", file: "/rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/panicking.rs", line: 502 }, { fn: "std::panicking::try", file: "/rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/panicking.rs", line: 466 }, { fn: "std::panic::catch_unwind", file: "/rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/panic.rs", line: 142 }, { fn: "std::rt::lang_start_internal::{{closure}}", file: "/rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/rt.rs", line: 148 }, { fn: "std::panicking::try::do_call", file: "/rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/panicking.rs", line: 502 }, { fn: "std::panicking::try", file: "/rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/panicking.rs", line: 466 }, { fn: "std::panic::catch_unwind", file: "/rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/panic.rs", line: 142 }, { fn: "std::rt::lang_start_internal", file: "/rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/rt.rs", line: 148 }, { fn: "std::rt::lang_start", file: "/rustc/cc66ad468955717ab92600c770da8c1601a4ff33/library/std/src/rt.rs", line: 165 }, { fn: "_main" }] }
```

Code: 

```
        let (config_filename, tokenizer_filename, weights_filename) = if self.offline {
            let cache = Cache::default().repo(repo);
            (
                cache
                    .get("config.json")
                    .ok_or(anyhow!("Missing config file in cache"))?,
                cache
                    .get("tokenizer.json")
                    .ok_or(anyhow!("Missing tokenizer file in cache"))?,
                cache
                    .get("pytorch_model.bin")
                    .ok_or(anyhow!("Missing weights file in cache"))?,
            )
        } else {
            let api = Api::new()?;
            let api = api.repo(repo);
            (
                api.get("config.json")?,
                api.get("tokenizer.json")?,
                api.get("pytorch_model.bin")?,
            )
        };
        let config = std::fs::read_to_string(config_filename)?;
        let config: Config = serde_json::from_str(&config)?;
        let tokenizer = Tokenizer::from_file(tokenizer_filename).map_err(E::msg)?;

        let vb = VarBuilder::from_pth(weights_filename, DTYPE, &device)?;
```

**Comment by LaurentMazare** [comment]
Thanks for reporting this, I just merged #1152 that should fix this. I also added a `--use-pth` flag to the bert example that you can use to switch to using the pytorch checkpoint rather than the safetensors file so that it's easier to test it out.

**Comment by liebkne** [comment]
@LaurentMazare That helps a lot. Thank you!
